alkira/resource_alkira_connector_gcp_interconnect_helper.go

```diff
@@ -137,71 +101,64 @@ func expandGcpInterconnectInstances(in []interface{}, so []interface{}, m interf
 		if v, ok := cfgInstance["vni_id"].(int); ok {
 			newInstance.Vni = v
 		}
-		segmentOptions, err := expandGcpInterconnectSegmentOptions(so, cfgInstance["name"].(string), m)
-		if err != nil {
-			return nil, err
+		if sO, ok := cfgInstance["segment_options"].([]interface{}); ok {
+			segmentOptions, err := expandGcpInterconnectSegmentOptions(sO, m)
+			if err != nil {
+				return nil, err
+			}
+			newInstance.SegmentOptions = segmentOptions
 		}
-		newInstance.SegmentOptions = segmentOptions
+
 		instances[i] = newInstance
 	}
 	return instances, nil
 }
 
-func setGcpInterconnectSegmentOptions(d *schema.ResourceData, instance *alkira.ConnectorGcpInterconnectInstance, m interface{}) ([]map[string]interface{}, error) {
+func setGcpInterconnectSegmentOptions(sO []alkira.ConnectorGcpInterconnectSegmentOption, m interface{}) ([]map[string]interface{}, error) {
```

Comment:
To make this simpler, you could just pass in instance and also fix the variable name.